Makefile

```diff
@@ -79,11 +81,15 @@ install:
 testbuild: TESTS := $(shell $(GO) list -tags '$(TAGS)' $(PKG))
 testbuild: GOFLAGS += -c
 testbuild:
+	ROOT="$(DIR)"; \
 	for p in $(TESTS); do \
 	  NAME=$$(basename "$$p"); \
 	  OUT="$$NAME.test"; \
-	  DIR=$$($(GO) list -f {{.Dir}} -tags '$(TAGS)' $$p); \
-	  $(GO) test $(GOFLAGS) -tags '$(TAGS)' -o "$$DIR"/"$$OUT" -ldflags '$(LDFLAGS)' "$$p" $(TESTFLAGS) || exit 1; \
+	  DIRROOT=$${ROOT-$$($(go) list -f {{.Root}} -tags '$(TAGS)' $$p)}; \
+	  OUTDIR=$$($(GO) list -f $${DIRROOT}/{{.ImportPath}} -tags '$(TAGS)' $$p); \
+	  $(GO) test $(GOFLAGS) -tags '$(TAGS)' -o "$$OUTDIR"/"$$OUT" -ldflags '$(LDFLAGS)' "$$p" $(TESTFLAGS) || exit 1; \
+	  # Silence output from strip, it complains when running on packages without tests. \
+	  if [ $${STRIPPED-0} -eq 1 ]; then strip -S "$$OUTDIR"/"$$OUT" >/dev/null 2>&1 || true; fi \
```

Comment:
or to put it another way: `1` and `2` are file descriptors pointing to `/dev/{stdout,stderr}`, redirects are executed in order and dereference pointers.